Userland/Libraries/LibWeb/Layout/GridFormattingContext.cpp

```diff
@@ -2091,13 +2091,38 @@ void GridFormattingContext::layout_absolutely_positioned_element(Box const& box)
     GridItem item { box, {}, {}, {}, {} };
     if (!is_auto_row) {
         auto row_placement_position = resolve_grid_position(box, GridDimension::Row);
-        item.row = row_placement_position.start;
-        item.row_span = row_placement_position.span;
+
+        // https://www.w3.org/TR/css-grid-2/#abspos
+        // 10.1. With a Grid Container as Containing Block
+        // If a grid-placement property refers to a non-existent line either by
+        // explicitly specifying such a line or by spanning outside of the existing
+        // implicit grid, it is instead treated as specifying auto (instead of
+        // creating new implicit grid lines).
+
+        int max_row = m_grid_rows.size();
+
+        // If the row start refers to a line before the first row or after the
+        // last row (intentionally allowing +1 row to allow placing abspos items
+        // between the last row and the container end), treat it as if it was `auto`.
+        auto start = row_placement_position.start;
+        if (start >= 0 && start <= max_row)
+            item.row = start;
+
+        // If the placement starts past the end it should ignore the grid entirely.
+        if (start <= max_row)
+            item.row_span = row_placement_position.span;
```

Comment:
I can try! Putting it in `resolve_grid_position()` was my first thought too, but I couldn't make it work cleanly. But that was also like 3 drafts ago, so maybe it'll work now? I'll give it a shot anyway.

Reply:
Just tried again, doing so requires a not-insignificant refactor as `GridFormattingContext::PlacementPosition` needs all its members made `optional`, and that keeps breaking things. I have a few other grid WPT fixes in the works that would make that process easier, but I was trying to keep this PR small. Could I leave this as a FIXME for now? Or should I merge those changes into this PR and make it bigger?

Reply:
I added a FIXME for now. If that's not acceptable let me know and I'll fold these changes into a larger PR with the other necessary supporting changes.

Reply:
yes, feel free to open bigger PR then because it's confusing to have position resolved once in `resolve_grid_position()` and then "fix" resolved position somewhere else.